receptorctl/receptorctl/socket_interface.py

```diff
@@ -98,7 +98,7 @@ def connect_to_service(self, node, service, tlsclient):
         if not str.startswith(text, "Connecting"):
             raise RuntimeError(text)
 
-    def submit_work(self, node, worktype, payload, tlsclient, ttl, params):
+    def submit_work(self, *, worktype, payload, node=None, tlsclient='', ttl='', params={}):
```

Comment:
AIUI `params={}` creates a new dict _once_ at module load time, and then uses it as the default.  So if params is ever modified by the function, the modification will stick around forever.  So we probably want to have the default be None and then test for that and instantiate a new dict in the function body (assuming the code can't deal with it being None).

Also, what does the `*` do?  I haven't seen this syntax before.

Reply:
Speaking of which, None is probably a better default than the empty string for tlsclient and (particularly) ttl.  And if the code can't deal with None values, it should be fixed, because someday someone will pass one in.

Reply:
> what does the * do?

https://www.python.org/dev/peps/pep-3102/

This syntax makes `worktype` and `payload` required keyword arguments, with no default value. 

This should illustrate it:

```
def myfunc(*, foo, bar):
    print(foo)
    print(bar)

myfunc()
```

Throws the error:

```
Traceback (most recent call last):
  File "test2.py", line 6, in <module>
    myfunc()
TypeError: myfunc() missing 2 required keyword-only arguments: 'foo' and 'bar
```

As an aside: if we ever wanted to add a new positional argument, it would need to be _before_ the `*`:

```
def myfunc(buzz, *, foo, bar):
    print(foo)
    print(bar)
    print(buzz)


myfunc('', foo='', bar='')
```

Reply:
Why do we want this?  If we just do

```
def submit_work(self, worktype, payload, node=None, tlsclient=None, ttl=None, params=None):
```
then if we ever want to add more parameters, that's fine - as long as they have defaults they won't break existing code.  I don't see any reason why we would ever need variadic arguments to this function.

Reply:
@ghjm Check out the latest diff. I've changed them all to `None`. But I much prefer keyword arguments over positional arguments. It makes reading the implementation code easier. It's possible to add whatever kinds of parameters we need to in the future, so I'm not sure I understand the concern.

Reply:
The concern is to reduce cognitive demands on API users, who are just using our library as one of hundreds they have to deal with as they go about their day.  Imagine if every library had a stylistic preference that they enforced on their callers.  I think we should stick to vanilla-looking Python that any user will easily understand and be able to use.